js/tutorial.js

```diff
@@ -20,48 +52,133 @@ Tutorial.prototype.constructor = Tutorial;
  ***********************************************/
 
 Tutorial.prototype.reset = function () {
-
-};
-
-Tutorial.prototype.draw = function (ctx) {
-    for (var i = 0, len = this.entities.length; i < len; i++) {
-        this.entities[i].draw(ctx);
-    }
+    
 };
 
 Tutorial.prototype.update = function () {
-    if (typeof this.entities !== 'undefined') {
-        var entitiesCount = this.entities.length;
-
-        // Cycle through the list of entities in Scene.
-        for (var i = 0; i < entitiesCount; i++) {
-            var entity = this.entities[i];
-
-            // Only update those not flagged for removal, for optimization
-            if (typeof entity !== 'undefined' && !entity.removeFromWorld) entity.update();
+    // Check for mouse/keyboard input
+    // game.mouse for hover, game.click for click
+    if (this.running) {
+        this.timeExpire -= this.game.clockTick;
+        if (this.timeExpire < 0) {
+            this.timeExpire = 15;
         }
-
-        // Removal of flagged entities
-        for (var j = this.entities.length - 1; j >= 0; --j) {
-            if (this.entities[j].removeFromWorld) this.entities.splice(j, 1);
+        this.charge += 1;
+        if (this.charge > 70) {
+            this.charge = 0;
+        }
+        
+        this.backButtonHover = false;
+        
+        if (this.game.mouse) {
+            if (this.game.mouse.x < this.backButtonHitbox.right && this.game.mouse.x > this.backButtonHitbox.left &&
+                this.game.mouse.y < this.backButtonHitbox.bottom && this.game.mouse.y > this.backButtonHitbox.top) {
+                
+                this.backButtonHover = true;
+            }
         }
+        
+        if (this.game.click) {
+            if (this.game.click.x < this.backButtonHitbox.right && this.game.click.x > this.backButtonHitbox.left &&
+                this.game.click.y < this.backButtonHitbox.bottom && this.game.click.y > this.backButtonHitbox.top) {
+                
+                this.isDone = true;;
+            }
+        }
+    }
+};
 
+Tutorial.prototype.draw = function (ctx) {
+    this.background.draw(ctx);
+    
+    // For a darker background
+    drawRoundedRect(ctx, 0, 0, 800, 600, 0, "rgba(0, 0, 0, 0.3)", "rgb(0, 0, 0)");
+    
+    // Back button
+    ctx.drawImage(ASSET_MANAGER.getAsset("./img/back.png"), 20, 490, 90, 90);
+   
+    // Objective
+    drawTextWithOutline(ctx, "26px Impact", "Objective:", 20, 40, 'indigo', 'white');
+    drawTextWithOutline(ctx, "22px Impact", "Gain points by being the sole goat on the hill!", 170, 40, 'black', 'white');
```

Comment:
Maybe above this line say `Fight for the [image of crown]!` to give a reason for gaining points.